### ISSUE TITLE
x/build: os.TestUserConfigDir and os.TestUserCacheDir failing on plan9-arm with `permission denied`

### ISSUE THREAD

**Issue by bcmills** [golang/go#62216]
For #57638, https://go.dev/cl/520262 added tests for a couple of previously-untested functions in the `os` package.
Those tests are currently failing on the `plan9-arm` builder with `permission denied` errors:
https://build.golang.org/log/6f395a2be0b4d0c7708c88cce47a863680a4eff1

The discussion on #29960 stated in no uncertain terms that `os.UserConfigDir` on Plan 9 should be `$home/lib`.

The tests verify that if the config directory can be located at all, the test can create a subdirectory within it. This is analogous to how a real program might store and read its own configuration data in a subdirectory of the config directory.

It seems likely to me that the builder is misconfigured in some way. I can think of three possible ways to resolve it:
- Perhaps `$home` should be set to a different directory (owned by the buildlet).
- Perhaps the permissions for the existing directory (`/usr/glenda`) and or buildlet shoud be adjusted so that the test has permission to write there.
- Perhaps the `$home` variable should be cleared entirely in the buildlet's environment, so that the test will know not to even try to write there.

(attn @millerresearch; CC @golang/plan9)

**Comment by millerresearch** [comment]
Yes, plan9-arm builders were configured in an overly-secure way which prevented the test user (glenda) writing to its own home directory. I'm surprised no test has revealed this before.As a workaround I've made writable lib and lib/cache directories under $workdir and bound these onto $home/lib and $home/lib/cache. That seems to make tests pass.

**Comment by bcmills** [comment]
Great, thank you!
